### PR TITLE
Investigate E2E CI caching issue

### LINKED ISSUE #11819: Playwright: tests passes locally and fails in CI. Investigate a potential caching issue
## Describe the bug

In #11029, we changed the Cart template's HTML title tag from h2 to h1. Subsequently, we adjusted the E2E tests in #11709 to reflect this change. However, while the test passes locally, it fails in GitHub's CI. It appears that in the CI environment, the Cart Template's title is still shown with an h2 tag instead of h1, possibly due to caching issues. We would like to delve deeper into this matter to understand why the CI is not reflecting the updated title tag.

## To reproduce

1. Restore the branch in #11709
2. Checkout to `update/cart-template-e2e-test` locally
3. Run the test `Template can be opened in the site editor` in `tests/e2e/tests/templates/cart-template.block_theme.spec.ts`
4. Ensure the test passes successfully
5. Re-run `Playwright Tests / Playwright E2E tests - Normal (pull_request)` in #11709. Ensure the same test fails

## Expected behavior

We should not have any caching issues in our CI environment.

### LINKED ISSUE #11820: Playwright E2E tests: Update the HTML tag of the Cart template title
## Describe the bug

In https://github.com/woocommerce/woocommerce-blocks/pull/11029, we changed the Cart template's HTML title tag from h2 to h1. Subsequently, we must adjust the E2E tests in `tests/e2e/tests/templates/cart-template.block_theme.spec.ts` to reflect this change.

Currently, the updated E2E tests fail in CI (see #11709). Before working on this ticket, we should fix the issue highlighted in #11819.

### PR DESCRIPTION
**Pull request by nielslange**
## What

Fixes woocommerce/woocommerce-blocks#11819
Fixes woocommerce/woocommerce-blocks#11820

## Why

In this PR, we address issues identified after our recent changes to the Page: Cart and Page: Checkout templates. Previously, in woocommerce/woocommerce-blocks#11029, we updated these pages' heading levels from h2 to h1, and in woocommerce/woocommerce-blocks#11709, we adjusted our E2E tests to align with these changes. However, we observed that the adjusted E2E tests were failing. Upon investigation, we identified the following problems:

- The welcome guide modal was not being closed in certain scenarios.
- With the WordPress 6.4 update, [the `Template options` button had been changed](https://github.com/WordPress/gutenberg/commit/bba4bbf8c998432159c6b1ea7db40bfe4fc23d5d).
- The page-content-wrapper was still loading post titles using the default h2 heading level, which contradicts our recent updates.

This PR aims to resolve these issues, ensuring that our E2E tests reflect the latest template modifications.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Please ensure that all Playwright E2E tests are passing.

* [x] Do not include in the Testing Notes 
* [x] Should be tested by the development team exclusively 

## Screenshots or screencast

n/a

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:

* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:

* [ ] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor.
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

n/a

### CODE REVIEW

File: tests/e2e/tests/templates/cart-template.block_theme.spec.ts [hunk 1]
```diff
@@ -17,17 +17,16 @@ test.describe( 'Test the cart template', async () => {
...
-				.locator( 'h2:has-text("Cart")' )
+				.locator( 'h1:has-text("Cart")' )
 				.first()
 		).toBeVisible();
 	} );
 
-	// Remove the skip once this ticket is resolved: https://github.com/woocommerce/woocommerce-blocks/issues/11671
-	// eslint-disable-next-line playwright/no-skipped-test
-	test.skip( 'Template can be accessed from the page editor', async ( {
+	test( 'Template can be accessed from the page editor', async ( {
```
Comment (tarhi-saad) [inline tests/e2e/tests/templates/cart-template.block_theme.spec.ts]: This test should be skipped. The #11671 was closed by mistake. It was automatically closed because it was mentioned in another PR.

File: tests/e2e/tests/templates/cart-template.block_theme.spec.ts [hunk 2]
```diff
@@ -37,13 +36,14 @@ test.describe( 'Test the cart template', async () => {
...
 		await editor.page.getByRole( 'button', { name: /Cart/i } ).click();
 		await editorUtils.enterEditMode();
+		await editorUtils.closeWelcomeGuideModal();
 		await expect(
-			editor.canvas.locator( 'h2:has-text("Cart")' ).first()
+			editor.canvas.locator( 'h1:has-text("Cart")' ).first()
 		).toBeVisible();
 		await editor.openDocumentSettingsSidebar();
-		await page.getByRole( 'button', { name: 'Edit template' } ).click();
+		await page.getByLabel( 'Template options' ).click();
```
Comment (tarhi-saad) [inline tests/e2e/tests/templates/cart-template.block_theme.spec.ts:50]: We should click the `Edit template` option after clicking on the `template options` label.

File: tests/e2e/tests/templates/checkout-template.block_theme.spec.ts [hunk 3]
```diff
@@ -25,9 +26,7 @@ test.describe( 'Test the checkout template', async () => {
 		).toBeVisible();
 	} );
 
-	// Remove the skip once this ticket is resolved: https://github.com/woocommerce/woocommerce-blocks/issues/11671
-	// eslint-disable-next-line playwright/no-skipped-test
-	test.skip( 'Template can be accessed from the page editor', async ( {
+	test( 'Template can be accessed from the page editor', async ( {
```
Comment (tarhi-saad) [inline tests/e2e/tests/templates/checkout-template.block_theme.spec.ts]: Same here! We should skip this test. This ticket is still relevant: #11671

**Review by tarhi-saad — CHANGES_REQUESTED** [review]
Thank you for working on this PR, @nielslange! The main issue is fixed. I added some comments below regarding the unskipped tests.

**Review by tarhi-saad — CHANGES_REQUESTED** [review]
@nielslange, could you fix the TS errors if the function `admin.visitSiteEditor()` in the Cart and Checkout templates. These errors were introduced in #11866.

**Comment by nielslange** [comment]
Thanks for your review, @tarhi-saad. I've address all your feedback and ensured that the tests are still passing. You can do another review now.

**Review by tarhi-saad — APPROVED** [review]
Thank you for the changes, @nielslange! 🙌 LGTM! 🚀
